### PR TITLE
feat(doc): Add a section about plugin documentation

### PR DESCRIPTION
**Pull request by loicmathieu**

### CODE REVIEW

File: docs/plugin-developer-guide/documentation/README.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+The following manifest attributes are used to document the group of plugins:
+
+- `X-Kestra-Title`: we usually use the Gradle `project.name` property.
+- `X-Kestra-Group`: we usually use the Gradle `group.id` property with an additional group name.
+- `X-Kestra-Description`: we usually use the Gradle `project.description` property.
+- `X-Kestra-Version`: we usually use the Gradle `project.version` property.
+
+If you follow the plugin structure of the GitHub template you should have something like that:
+
+```ruby
```
Comment (tchiotludo) [inline docs/plugin-developer-guide/documentation/README.md:31]:
```suggestion
```groovy
```

**Review by tchiotludo — CHANGES_REQUESTED** [review]

File: docs/plugin-developer-guide/documentation/README.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+@NotNull
+@Schema(
+    title = "The source of the published data.",
+    description = "Can be an internal storage URI, a list of Pub/Sub messages, or a single Pub/Sub message.",
+    anyOf = {String.class, Message[].class, Message.class}
+)
+private Object from;
+```
+
+Note that, due to limitation on the way JSON schema works, you cannot add `@Schema` both on a Java enum type and the plugin property that uses this type. We advise to avoid using `@Schema` on enumerations.
```
Comment (tanay1337) [inline docs/plugin-developer-guide/documentation/README.md:155]:
```suggestion
**Note:** Due to limitations on how JSON schema works, you cannot add `@Schema` on a Java enum type and the plugin property that uses this type. We advise avoiding using `@Schema` on enumerations.
```

File: docs/plugin-developer-guide/documentation/README.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+### Document the plugin outputs
+
+Outputs should be documented with the `io.swagger.v3.oas.annotations.media.Schema` annotation the same way as plugin properties, please read the section above for more information.
```
Comment (tanay1337) [inline docs/plugin-developer-guide/documentation/README.md:159]:
```suggestion
Outputs should be documented with the `io.swagger.v3.oas.annotations.media.Schema` annotation the same way as plugin properties. Please read the section above for more information.
```

File: docs/plugin-developer-guide/documentation/README.md [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+ 
+Only this annotation should be used, never use `@PluginProperty` on an output.
```
Comment (tanay1337) [inline docs/plugin-developer-guide/documentation/README.md:161]:
```suggestion
Only use the annotation mentioned above. Never use `@PluginProperty` on an output.
```

**Review by tanay1337 — CHANGES_REQUESTED** [review]

File: docs/plugin-developer-guide/documentation/README.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
+---
+order: 8
+---
+# Document your plugin
+
+First, let remember ourself the organization of a plugin project:
+
+- The Gradle project is called a group of plugins as it contains multiple plugins.
+- The package in which a plugin is, is called a sub-group of plugins, sometimes there is only one sub-group so the group and the sub-group are the same.
+- Each class is a plugin that provides one of: task, trigger, condition, ...
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply docs/plugin-developer-guide/documentation/README.md:10]: Yes, a Java class extends a base class that is the `Task` class, the `Trigger` class, the `Condition` task, ...

File: docs/plugin-developer-guide/documentation/README.md [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+                "X-Kestra-Title": project.name,
+                "X-Kestra-Group": project.group + ".gcp",
+                "X-Kestra-Description": project.description,
+                "X-Kestra-Version": project.version
+        )
+    }
+}
+```
+
+As you can see, the most important documentation attribute is the `description` that should be a short sentence that describe the group of plugins.
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply docs/plugin-developer-guide/documentation/README.md:49]: I think we should keep 'group of plugins' here

File: docs/plugin-developer-guide/documentation/README.md [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+The `@PluginProperty` annotation contains two attributes:
+
+- `dynamic`: set it to true if the property will be rendered dynamically, see [Render Variable](..//runnable-task/#render-variables).
+- `additionalProperties`: you can use it to denote the sub-type of the property, for example when using a `Map<String, Message>` you can set it to `Message.class`.
+
+The Swagger `@Schema` annotation contains a lot of attributes that can be used to document the plugin properties, the most useful are:
+
+- `title`: short description of a property.
+- `description`: long description of a property.
+- `anyOf`: allows to specify which types are allowed when the type of the property can be extended (interface, abstract classes, ...). This should be set when the type of the property is `Object`.
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply docs/plugin-developer-guide/documentation/README.md:138]: I had a hard time explaining this ...
I think the best would be to describe more precisely how to use it

> specifying which sub-types are allowed when the property type is an interface, an abstract class, or a class inside a hierarchy of classes.

File: docs/plugin-developer-guide/documentation/README.md (hunk 2 above)
Comment (loicmathieu) [reply docs/plugin-developer-guide/documentation/README.md:155]: I'm pretty sure we have a format to display notes as a special visual block, I'll check and update the formating.

File: docs/plugin-developer-guide/documentation/README.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
+---
+order: 8
+---
+# Document your plugin
+
+First, let remember ourself the organization of a plugin project:
+
+- The Gradle project is called a group of plugins as it contains multiple plugins.
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply docs/plugin-developer-guide/documentation/README.md:8]: > - The Gradle project can contain several plugins, we call it a group of plugins.

File: docs/plugin-developer-guide/documentation/README.md [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+
+## Document the plugin group
+
+### Manifest attributes
+
+Kestra uses custom manifest attributes to provides group top level documentation.
+
+The following manifest attributes are used to document the group of plugins:
+
+- `X-Kestra-Title`: we usually use the Gradle `project.name` property.
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply docs/plugin-developer-guide/documentation/README.md:24]: Those are defaults set by the plugin template repository. I can replace each by:

> by default, the Gradle `project.name` property is used.

File: docs/plugin-developer-guide/documentation/README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (tanay1337) [reply docs/plugin-developer-guide/documentation/README.md]: Okay, sounds good.

File: docs/plugin-developer-guide/documentation/README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (tanay1337) [reply docs/plugin-developer-guide/documentation/README.md:24]: Sounds good.

File: docs/plugin-developer-guide/documentation/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (tanay1337) [reply docs/plugin-developer-guide/documentation/README.md:8]: That sounds good!

**Comment by loicmathieu** [comment]
@tanay1337 thanks a lot for your review, I apply all your suggestions and update the part that was still open for discussion. Car you do a quick check before we can merge it ?

**Comment by tanay1337** [comment]
LGTM @loicmathieu 🙌

**Comment by loicmathieu** [comment]
@tchiotludo can I merge it ?

**Pull request merged**